tests/spm/SPM_scenario9Context.hpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * @file SPM_scenario9Context.hpp
+ * @author S. Martorana (salvatoremartorana@hotmail.com)
+ * @author F. Abrignani (federignoli@hotmail.it)
+ * @brief Alias SPM_scenario9Context.
+ * @version 0.1
+ * @date 2023-10-06
+ * @private
+ * @copyright Copyright (c) 2023 Federico Abrignani (federignoli@hotmail.it).
+ * Distributed under the Boost Software License, Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt).
+ * 
+ */
+
+#include <boost/fairness.hpp>
+#include <DeterministicConcurrency>
+#include <vector>
+#include <random>
+#include <mutex>
+
+namespace SPM_scenario9{
+    using namespace DeterministicConcurrency;
+
+    boost::fairness::shared_priority_mutex<7> m;
+    std::mutex sm;
+
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<int> distribution(0, 6);
+
+    void threadFunction(thread_context* c ,int i) {
+        c->lock_shared(&m, i);
+        c->switchContext();
+        m.unlock_shared();
+    }
+
+    void controlThread(thread_context* c) {
+        c->lock(&m);
+        c->switchContext();
+        m.unlock();
+    }
+
+    static size_t THREAD0 = 0;
+    static size_t THREAD1 = 1;
+    static size_t THREAD2 = 2;
+    static size_t THREAD3 = 3;
+    static size_t CTRLTHREAD = 4;
+        
+    static constexpr auto executeSchedulingSequence = []{
+        auto thread_0 = std::tuple{&threadFunction, distribution(gen)};
+        auto thread_1 = std::tuple{&threadFunction, distribution(gen)};
+        auto thread_2 = std::tuple{&threadFunction, distribution(gen)};
+        auto thread_3 = std::tuple{&threadFunction, distribution(gen)};
+        auto ctrlThread = std::tuple{&controlThread};
+        
+        auto sch = make_UserControlledScheduler(
+            thread_0, thread_1, thread_2, thread_3, ctrlThread
+        );
+
```

Comment:
Delete these comments and just say at line 64
// if the lock_shared wouldn`t be "shared" this instruction would cause deadlock.